geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -2125,6 +2125,10 @@ public void close(String reason, Throwable systemFailureCause, boolean keepAlive
 
         // do this before closing regions
         resourceManager.close();
+        meterRegistry.close();
+
+        for (MeterRegistry tmpMeterRegistry : meterSubregistries)
+          tmpMeterRegistry.close();
```

Comment:
I would hesitate to close registries Geode didn't install. Consider a client application providing its own registry that records meters for itself and wants Geode recorded to the same registry. Now as the cache is closed so will the user's registry and their application meters will not record anymore.

Reply:
Perhaps I am not understanding... If the cache is closed, isn't the result same? This just cleans up everything when the cache goes away. If we don't do that, I think they might leak memory like we were doing before.

Reply:
So are you proposing to pull the closure of submeterregistries? If so, I have done that it appears fine. Partly because there appears to be no submeterregistries.